src/vos/vos_tree.c

```diff
@@ -937,10 +937,12 @@ key_tree_prepare(struct vos_object *obj, daos_handle_t toh,
 		break;
 	}
 
-	if (sub_toh)
+	if (sub_toh) {
+		D_ASSERT(krec != NULL);
 		rc = tree_open_create(obj, tclass, flags, krec, sub_toh);
-	if (rc)
+	} if (rc) {
```

Comment:
(style)  trailing statements should be on next line (or did you mean 'else if'?)